### DISCUSSION TITLE
Fido2 / Passkey support

### DISCUSSION THREAD

**Discussion by nimchak** [pnp/powershell#4354] [General]
It is imperative to add support for logging in with Fido2/Passkeys.  Currently this does not seem to work.  Many organizations are forcing admins to utilize Fido2 keys for login these days.

**Comment by gautamdsheth** [comment]
This is now supported when you use connect-pnponline with oslogin and clientid parameter.

**Comment by nimchak** [comment]
I see the reference to FIDO support but are you sure FIDO2/Passkeys are supported too?  I have been trying to get it to work with no luck.

**Reply by gautamdsheth** [reply]
What is the issue ? What error are you seeing when you try it ?

**Reply by gautamdsheth** [reply]
We use Msal under the hood , if it’s supported with Msal it should work here :

https://learn.microsoft.com/en-us/entra/msal/dotnet/acquiring-tokens/desktop-mobile/wam
